src/transport/SecureSession.h

```diff
@@ -241,6 +241,14 @@ class SecureSession : public Session, public ReferenceCounted<SecureSession, Sec
     // It requires that the target sessoin is also a CASE session, having the same peer and CATs as this session.
     void NewerSessionAvailable(const SessionHandle & session);
 
+#if CHIP_PROGRESS_LOGGING
```

Comment:
Please just leave this unconditionally. The cost is extremely minimal but having the conditionality causes multiple complex formats to be accounted for slightly differently. I think the format should outright have the %d rather than going to through this indirection via a temporary string

Reply:
I can leave this unconditional, but note that the callsites are all conditional on CHIP_PROGRESS_LOGGING.

I guess that means this will be dead-code-stripped, so that's fine.

Reply:
> I think the format should outright have the %d rather than going to through this indirection via a temporary string

Non-secure sessions don't have session ids.  And there is no reserved session id value to indicate "not a session".  So we need the indirection so we don't log bogus and misleading session ids for non-secure sessions, no?  @tcarmelveilleux